File: src/SSHServer/Packets/Disconnect.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SSHServer.Packets
{
    public class Disconnect : Packet
    {
        public override PacketType PacketType
        {
            get
            {
                return PacketType.SSH_MSG_DISCONNECT;
            }
        }

        public DisconnectReason Reason { get; set; }
        public string Description { get; set; }
        public string Language { get; set; } = "en";

        public override void Load(ByteReader reader)
        {
            Reason = (DisconnectReason)reader.GetUInt32();
            Description = reader.GetString(Encoding.UTF8);
            if (!reader.IsEOF)
                Language = reader.GetString();
        }

        protected override void InternalGetBytes(ByteWriter writer)
        {
            writer.WriteUInt32((uint)Reason);
            writer.WriteString(Description, Encoding.UTF8);
            writer.WriteString(Language);
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Packets/Disconnect.ts
import { Packet } from "./Packet";
import { PacketType } from "./PacketType";
import { ByteReader } from "../ByteReader";
import { ByteWriter } from "../ByteWriter";
import { DisconnectReason } from "./DisconnectReason";

export class Disconnect extends Packet {
    public reason: DisconnectReason;
    public description: string;
    public language: string = "en";

    public getPacketType(): PacketType {
        return PacketType.SSH_MSG_DISCONNECT;
    }

    protected internalGetBytes(writer: ByteWriter) {
        writer.writeUInt32(<number>this.reason);
        writer.writeString(this.description, "UTF8");
        writer.writeString(this.language);
    }

    public load(reader: ByteReader) {
        this.reason = <DisconnectReason>reader.getUInt32();
        this.description = reader.getString("UTF8");
        if (!reader.isEOF())
            this.language = reader.getString();
    }
}
